password_change_restrict/models/res_users.py

```diff
@@ -0,0 +1,48 @@
+# -*- coding: utf-8 -*-
+# License and authorship information in:
+# __openerp__.py file at the root folder of this module.
+
+from openerp import api, exceptions, models, SUPERUSER_ID, _
+
+
+class ResUsers(models.Model):
+    _inherit = 'res.users'
+
+    SELF_WRITEABLE_FIELDS = ['signature', 'action_id', 'company_id', 'email',
```

Comment:
that shouldn't matter too much in this case. Generally, `_register_hook` is safer because your model is fully loaded (relevant if you need to look up stuff in the db, patch fields, etc)